### PR TITLE
Add deprecation notes, update to Gradle 7.6

### PR DESCRIPTION
**Pull request by EnnuiL**
This paves the way for the eventual removal of QMoL on Quilt Loom

### CODE REVIEW

**Review by IotaBread — CHANGES_REQUESTED** [review]
There are class files here?

**Comment by EnnuiL** [comment]
> There are class files here?

VSCode has a `bin` folder and I forgot that some .gitignore files are incomplete

**Review by IotaBread — APPROVED** [review]

File: test/build.gradle.kts [hunk 1]
```diff
@@ -14,9 +14,10 @@ repositories {
     }
 }
 
-var minecraft_version = "1.18.2-rc1"
-var quilt_mappings = "1.18.2-rc1+build.1"
-var loader_version = "0.12.2"
+var minecraft_version = "1.18.2"
+// Test QMoL crashing with post-Intermediary publication builds by swtichting to 1.18.2+build.26
```
Comment (ix0rai) [inline test/build.gradle.kts:18]:
```suggestion
// QMoL crashing with post-Intermediary publication builds can be tested by switching to 1.18.2+build.26
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -2,16 +2,16 @@ name: Release
...
+      - uses: gradle/wrapper-validation-action@v1
       - run: ./gradlew build publish --stacktrace
         env:
-          MAVEN_URL: ${{ secrets.MAVEN_URL }}
-          MAVEN_USERNAME: ${{ secrets.MAVEN_USERNAME }}
-          MAVEN_PASSWORD: ${{ secrets.MAVEN_PASSWORD }}
\ No newline at end of file
+            MAVEN_URL: ${{ secrets.MAVEN_URL }}
+            MAVEN_USERNAME: ${{ secrets.MAVEN_USERNAME }}
+            MAVEN_PASSWORD: ${{ secrets.MAVEN_PASSWORD }}
```
Comment (ix0rai) [inline .github/workflows/release.yml:17]: unnecessary whitespace changes

**Review by eristhea — APPROVED** [review]
nitpicks, but this looks good!

File: .github/workflows/release.yml (hunk 2 above)
Comment (EnnuiL) [reply .github/workflows/release.yml:17]: I was going to say something about consistency between projects, but nope! i probably just screwed up here; Will fix

Comment (EnnuiL) [reply .github/workflows/release.yml]: Fixed!

File: src/main/java/org/quiltmc/quiltmappings/loom/QuiltMappingsOnLoomPlugin.java [hunk 3]
```diff
@@ -110,6 +116,10 @@ public void visit(MappingVisitor mappingVisitor) throws IOException {
             if (!intermediaryToQm.exists()) {
                 List<File> quiltMappings = new ArrayList<>(project.getConfigurations().detachedConfiguration(project.getDependencies().create(this.quiltMappings)).resolve());
 
+                if (quiltMappings.size() < 2) {
+                    throw new UnsupportedOperationException("The Quilt Mappings build used here is likely to be a post-Intermediary publications build, which are incompatible with Quilt Mappings on Loom!");
```
Comment (OroArmor) [inline src/main/java/org/quiltmc/quiltmappings/loom/QuiltMappingsOnLoomPlugin.java:120]: I would add examples of what to put for both layered and non layered mappings here

**Review by OroArmor — CHANGES_REQUESTED** [review]

File: src/main/java/org/quiltmc/quiltmappings/loom/QuiltMappingsOnLoomPlugin.java (hunk 3 above)
Comment (EnnuiL) [reply src/main/java/org/quiltmc/quiltmappings/loom/QuiltMappingsOnLoomPlugin.java:120]: Maybe linking to QTM's examples would work?

Comment (OroArmor) [reply src/main/java/org/quiltmc/quiltmappings/loom/QuiltMappingsOnLoomPlugin.java:120]: Sure!

**Review by OroArmor — APPROVED** [review]
